x-pack/plugins/uptime/public/pages/monitor.tsx

```diff
@@ -43,6 +44,10 @@ export const MonitorPage: React.FC = () => {
 
   const monitorId = useMonitorId();
 
+  useEffect(() => {
+    dispatch(setSelectedMonitorId(monitorId));
```

Comment:
Why do we need the UI reducer when we have `useMonitorId`? This could use a comment because I assume it's related to alerting etc. but it's hard to see the why here.

Reply:
this is required for alert editing, we don't have url state in alert app. We are only using redux there. So this is why we need it in store as well. I will add the comment.